packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/DestinationNetworkBox.tsx

```diff
@@ -24,158 +19,207 @@ import {
 } from '../../common/NetworkSelectionContainer'
 import { useNativeCurrencyBalances } from './useNativeCurrencyBalances'
 import { useIsBatchTransferSupported } from '../../../hooks/TransferPanel/useIsBatchTransferSupported'
-import { ether } from '../../../constants'
+import { getBridgeUiConfigForChain } from '../../../util/bridgeUiConfig'
+import { SafeImage } from '../../common/SafeImage'
+import { useTokensFromLists, useTokensFromUser } from '../TokenSearchUtils'
 import { formatAmount } from '../../../util/NumberUtils'
 import { Loader } from '../../common/atoms/Loader'
 import { useAmount2InputVisibility } from './SourceNetworkBox'
-import { useIsCctpTransfer } from '../hooks/useIsCctpTransfer'
 import { useArbQueryParams } from '../../../hooks/useArbQueryParams'
+import { useIsCctpTransfer } from '../hooks/useIsCctpTransfer'
 
-function NativeCurrencyDestinationBalance({ prefix }: { prefix?: string }) {
-  const nativeCurrencyBalances = useNativeCurrencyBalances()
+function BalanceRow({
+  parentErc20Address,
+  balance,
+  symbolOverride
+}: {
+  parentErc20Address?: string
+  balance: string | undefined
+  symbolOverride?: string
+}) {
   const [networks] = useNetworks()
-  const nativeCurrency = useNativeCurrency({
-    provider: networks.destinationChainProvider
-  })
-  const { isDepositMode } = useNetworksRelationship(networks)
-
-  if (nativeCurrency.isCustom) {
-    return (
-      <TokenBalance
-        forToken={nativeCurrency}
-        balance={nativeCurrencyBalances.destinationBalance}
-        on={isDepositMode ? NetworkType.childChain : NetworkType.parentChain}
-        prefix={prefix}
-      />
-    )
-  }
-  if (!nativeCurrencyBalances.destinationBalance) {
-    return (
-      <p className="flex items-center gap-1">
-        <span className="font-light">{prefix}</span>
-        <Loader color="white" size="small" />
-      </p>
-    )
-  }
+  const { childChainProvider, isDepositMode } =
+    useNetworksRelationship(networks)
+  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+
+  const tokensFromLists = useTokensFromLists()
+  const tokensFromUser = useTokensFromUser()
+
+  const tokenLogoSrc = useMemo(() => {
+    if (parentErc20Address) {
+      return (
+        tokensFromLists[parentErc20Address]?.logoURI ??
+        tokensFromUser[parentErc20Address]?.logoURI
+      )
+    }
+
+    return nativeCurrency.logoUrl
+  }, [
+    nativeCurrency.logoUrl,
+    parentErc20Address,
+    tokensFromLists,
+    tokensFromUser
+  ])
+
+  const symbol = useMemo(() => {
+    if (symbolOverride) {
+      return symbolOverride
+    }
+
+    if (parentErc20Address) {
+      return (
+        tokensFromLists[parentErc20Address]?.symbol ??
+        tokensFromUser[parentErc20Address]?.symbol
+      )
+    }
+
+    return nativeCurrency.symbol
+  }, [
+    symbolOverride,
+    nativeCurrency.symbol,
+    parentErc20Address,
+    tokensFromLists,
+    tokensFromUser
+  ])
```

Comment:
this is not working for Arb One/Sepolia native USDC -> Orbit chain

note the empty token symbol

can you check if you need the usage of `sanitizeTokenSymbol` because it handles all these cases and was the reason why we had it in the first place

<img width="673" alt="image" src="https://github.com/user-attachments/assets/144ccc33-32d7-40ab-8693-42ab3ab6fe09" />